Patch version update for backwards incompatibe change

Hi,

From version "0.3.2" to "0.3.3" a backwards incompatible change was made to the API:

```
error[E0027]: pattern does not mention field `kind`
   --> src/vmm.rs:424:17
    |
424 | /                 userfaultfd::Event::Pagefault {
425 | |                     rw: _,
426 | |                     addr,
427 | |                 } => addr as usize,
    | |_________________^ missing field `kind`
```

That shouldn't be done as per Cargo semver specification: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html ; changes to the patch version should be backwards compatible.

Please can you recall "0.3.3" and rather release "0.4.0"?

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "userfaultfd"
-version = "0.3.3"
+version = "0.4.0"
 authors = ["Adam C. Foltzer <acfoltzer@fastly.com>"]
 edition = "2018"
 license = "MIT OR Apache-2.0"

